test/latlonbound_test_utilities.py

```diff
@@ -0,0 +1,386 @@
+        angle_v1_v2_rad = uxarray.helpers.angle_of_2_vectors(v_b, v_c)
+        v0 = np.array(uxarray.helpers._normalize_in_place(np.cross(np.cross(v1_cart, v2_cart), v_b)))
+        avg_angle_rad = angle_v1_v2_rad / 10
+
+        for i in range(0, 10):
+            angle_rad_prev = avg_angle_rad * i
+            if i >= 9:
+                angle_rad_next = angle_v1_v2_rad
+            else:
+                angle_rad_next = angle_rad_prev + avg_angle_rad
+
+            # Get the two vectors of this section
+            w1_new = np.cos(angle_rad_prev) * v_b + np.sin(angle_rad_prev) * v0
+            w2_new = np.cos(angle_rad_next) * v_b + np.sin(angle_rad_next) * v0
+
+            # convert the 3D [x, y, z] vector into 2D lat/lon vector
+            w1_lonlat = uxarray.helpers._convert_node_xyz_to_lonlat_rad(list(w1_new))
+            w2_lonlat = uxarray.helpers._convert_node_xyz_to_lonlat_rad(list(w2_new))
+
+            # Manually set the left and right boundaries to avoid error accumulation
+            if i == 0:
+                w1_lonlat[1] = b_lonlat[1]
+            elif i >= 9:
+                w2_lonlat[1] = c_lonlat[1]
+
+            max_lat = max(max_lat, w1_lonlat[1], w2_lonlat[1])
+
+            if np.absolute(w2_lonlat[1] -
+                           w1_lonlat[1]) <= err_tolerance or w1_lonlat[
+                1] == max_lat == w2_lonlat[1]:
+                max_section = np.array([w1_new, w2_new])
+                break
+
+            # if the largest absolute value of lat at each sub-interval point b_i.
+            # Repeat algorithm with the sub-interval points (b,c)=(b_{i-1},b_{i+1})
+            if np.absolute(max_lat - w1_lonlat[1]) <= err_tolerance:
+                if i != 0:
+                    angle_rad_prev -= avg_angle_rad
+                    w1_new = np.cos(angle_rad_prev) * v_b + np.sin(angle_rad_prev) * v0
+                    w2_new = np.cos(angle_rad_next) * v_b + np.sin(angle_rad_next) * v0
+                    max_section = [w1_new, w2_new]
+                else:
+                    max_section = [v_b, w2_new]
+
+            elif np.absolute(max_lat - w2_lonlat[1]) <= err_tolerance:
+                if i != 9:
+                    angle_rad_next += avg_angle_rad
+                    w1_new = np.cos(angle_rad_prev) * v_b + np.sin(angle_rad_prev) * v0
+                    w2_new = np.cos(angle_rad_next) * v_b + np.sin(angle_rad_next) * v0
+                    max_section = [w1_new, w2_new]
+                else:
+                    max_section = [w1_new, v_c]
+
+        b_lonlat = uxarray.helpers._convert_node_xyz_to_lonlat_rad(copy.deepcopy(max_section[0]))
+        c_lonlat = uxarray.helpers._convert_node_xyz_to_lonlat_rad(copy.deepcopy(max_section[1]))
+
+    return np.average([b_lonlat[1], c_lonlat[1]])
+
+
+def min_latitude_rad(v1, v2):
+    """Quantitative method to find the minimum latitude between in a great circle arc recursively
```

Comment:
The function as written here isn't recursive, is it meant to be?

Reply:
Yes, it's meant to be non-recursive. 

Because python actually doesn't support recursion that well and might cause stack over flow easily under some marginal cases